src/multiplex/client.rs

```diff
@@ -138,21 +241,138 @@ where
     }
 }
 
-// ===== ClientInner =====
+// ===== Pending =====
 
-struct Pending<Tag, Item> {
+/// A type used to track in-flight requests.
+///
+/// Each pending response contains an associated `Tag` that is provided
+/// from the `TagStore` used to uniquely identify a request/response pair.
+pub struct Pending<Tag, Response> {
     tag: Tag,
-    tx: tokio::sync::oneshot::Sender<ClientResponse<Item>>,
+    tx: tokio::sync::oneshot::Sender<ClientResponse<Response>>,
     span: tracing::Span,
 }
 
+impl<Tag, Response> Pending<Tag, Response> {
+    /// Get the tag associated with this pending request.
+    pub fn tag(&self) -> &Tag {
+        &self.tag
+    }
+}
+
+impl<Tag, Response> fmt::Debug for Pending<Tag, Response>
+where
+    Tag: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Pending")
+            .field("tag", &self.tag)
+            .field("span", &self.span)
+            .finish()
+    }
+}
+
+// ===== Builder =====
+
+/// The default service error handler.
+pub type DefaultOnServiceError<E> = Box<dyn FnOnce(E) + Send>;
+
+/// Builder for [`Client`] this is used to configure the transport, pending store
+/// and service_handler.
+///
+/// # Defaults
+///
+/// By default this builder only requires a transport and sets a default [`PendingStore`]
+/// and error handler. The default pending store is just a [`VecDeque`] and the default
+/// error handler is just an empty closure.
+pub struct Builder<
+    T,
+    E,
+    Request,
+    P = VecDequePendingStore<T, Request>,
+    F = DefaultOnServiceError<E>,
```

Comment:
I _think_ I would flip these, since users are probably more likely to override the `F` than the `P`.